portal-frontend/src/app/shared/owner-dialogs/parcel-owners/parcel-owners.component.ts

```diff
@@ -92,13 +93,17 @@ export class ParcelOwnersComponent {
       });
     }
     dialog.beforeClosed().subscribe((updatedUuid) => {
-      if (updatedUuid) {
+      if (updatedUuid?.ownerDeleted) {
+        this.onOwnersDeleted.emit(updatedUuid.deletedOwnerId);
+      }
+      else if (updatedUuid) {
         this.onOwnersUpdated.emit();
       }
     });
     dialog.afterClosed().subscribe((result) => {
       if (result.ownerDeleted) {
-        this.dataSource.data = this.dataSource.data.filter((owner: ApplicationOwnerDto | NoticeOfIntentOwnerDto) => owner.uuid !== result.deletedOwnerId);
+        const updatedOwners = this.dataSource.data.filter((o: ApplicationOwnerDto | NoticeOfIntentOwnerDto) => o.uuid !== result.deletedOwnerId);
```

Comment:
It's not great practice to mimic what the API call is doing in the front end, as this way if the backend were to fail or not delete the owner, it would appear in the front end as though it worked. Can you re-load the owners after the delete instead?

Reply:
removed it, kept the logic as is